server/game/cards/07-GR/MarketFluctuation.js

```diff
@@ -0,0 +1,27 @@
+const Card = require('../../Card.js');
+
+class MarketFluctuation extends Card {
+    // Play: Each haunted player draws 3 cards. Each player that is
+    // not haunted discards 3 random cards from their hand.
+    setupCardAbilities(ability) {
+        this.play({
+            gameAction: ability.actions.draw((context) => ({
```

Comment:
I think you could use `ability.actions.sequential` instead of `then` with `allwaysTriggers`. Anyway, it has the same result.

Reply:
Done, thanks.